Mathlib/Order/Hom/WithTopBot.lean

```diff
@@ -63,6 +63,34 @@ def coeOrderHom {α : Type*} [Preorder α] : α ↪o WithTop α where
   inj' := WithTop.coe_injective
   map_rel_iff' := WithTop.coe_le_coe
 
+/-- Removing then adding ⊤ makes the type `OrderIso` to the original type. -/
+def subtypeOrderIso [DecidableEq α] [PartialOrder α] [OrderTop α] :
+    WithTop {a : α // a ≠ ⊤} ≃o α where
+  toFun
+  | .some a => a
+  | ⊤ => ⊤
```

Comment:
You might want to use [WithTop.untopD](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/WithBot.html#WithTop.untopD) for this part.